pkg/apis/nmstate/v1/state_test.go

```diff
@@ -0,0 +1,103 @@
+package v1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	yaml "github.com/ghodss/yaml"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	var (
+		obtainedStruct   NodeNetworkState
+		obtainedManifest []byte
+		desiredState     = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up`)
+
+		currentState = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: down`)
+
+		NNSManifest = `
+apiVersion: nmstate.io/v1
+kind: NodeNetworkState
+metadata:
+  name: node01
+  creationTimestamp: "1970-01-01T00:00:00Z"
+spec:
+  managed: true
+  nodeName: node01
+  desiredState:
+    interfaces:
+      - name: eth1
+        type: ethernet
+        state: up
+currentState:
+  interfaces:
+    - name: eth1
+      type: ethernet
+      state: down
+
+`
+		NNSStruct = NodeNetworkState{
```

Comment:
we don't need obtainedStruct intantiation if we have the definition right here

Reply:
We use this at both marshal and unmarshal so we need the global definition